src/arithmetic/algebraic_expression/algebraic_expression_construction.c

```diff
@@ -329,81 +331,73 @@ static AlgebraicExpression *_AlgebraicExpression_OperandFromEdge
 	ASSERT(e != NULL);
 
 	uint reltype_id;
-	QGNode               *src_node          =  e->src;
-	QGNode               *dest_node         =  e->dest;
-	AlgebraicExpression  *add               =  NULL;
-	AlgebraicExpression  *root              =  NULL;
-	AlgebraicExpression  *src_filter        =  NULL;
-	bool                 var_len_traversal  =  QGEdge_VariableLength(e);
+	QGNode              *src_node         = e->src;
+	QGNode              *dest_node        = e->dest;
+	AlgebraicExpression *add              = NULL;
+	AlgebraicExpression *root             = NULL;
+	AlgebraicExpression *src_filter       = NULL;
+	bool                var_len_traversal = QGEdge_VariableLength(e);
 
 	// use original `src` and `dest` for algebraic operands
-	const  char  *src   =  (transpose)               ?  dest_node->alias  :  src_node->alias;
-	const  char  *dest  =  (transpose)               ?  src_node->alias   :  dest_node->alias;
-	const  char  *edge  =  _should_populate_edge(e)  ?  e->alias          :  NULL;
+	const char *src  = (transpose)              ? dest_node->alias : src_node->alias;
+	const char *dest = (transpose)              ? src_node->alias  : dest_node->alias;
+	const char *edge = _should_populate_edge(e) ? e->alias         : NULL;
 
 	// if src node has a label, multiply to the left by label matrix
 	if(QGNode_LabelCount(src_node) > 0) {
 		src_filter = _AlgebraicExpression_OperandFromNode(src_node);
 	}
 
-	// no hops: (a)-[:R*0]->(b)
-	// in this case we want to use the identity matrix
-	// f * I  = f 
-	if(!var_len_traversal && e->minHops == 0) {
-		root = AlgebraicExpression_NewOperand(IDENTITY_MATRIX, true, src, dest,
-				edge, "I");
-	} else {
-		uint reltype_count = array_len(e->reltypeIDs);
-		switch(reltype_count) {
-		case 0: // no relationship types specified; use the adjacency matrix
-			root = AlgebraicExpression_NewOperand(NULL, false, src, dest, edge, NULL);
-			break;
-		case 1: // single relationship type
-			root = AlgebraicExpression_NewOperand(NULL, false, src, dest, edge, e->reltypes[0]);
-			break;
-		default: // multiple edge type: -[:A|:B]->
-			add = AlgebraicExpression_NewOperation(AL_EXP_ADD);
-			for(uint i = 0; i < reltype_count; i++) {
-				AlgebraicExpression *operand = AlgebraicExpression_NewOperand(
-						NULL, false, src, dest, edge, e->reltypes[i]);
-				AlgebraicExpression_AddChild(add, operand);
-			}
-			root = add;
-			break;
+	uint reltype_count = array_len(e->reltypeIDs);
```

Comment:
so now we don't use IDENTITY_MATRIX anymore maybe we can delete it?

Reply:
we sure can